### DISCUSSION TITLE
Bug: The effect of PIXI.Graphics is not the same as canvas

### DISCUSSION THREAD

**Discussion by ymma** [pixijs/pixijs#9080] [Q&A] [answered]
### Current Behavior

I want to display some graphics from svg, but I have a problem.
The following are the effects of canvas and PixiJS respectively, they are not the same.
I want to know how to use PixiJS to achieve the effect on the canvas

<img width="396" alt="image" src="https://user-images.githubusercontent.com/10007630/213018217-1cf7eeae-f844-47f9-89a4-aba08f27cb85.png">

### Expected Behavior

<img width="392" alt="image" src="https://user-images.githubusercontent.com/10007630/213018292-7300210e-ec90-4e4b-9498-7d2325854c2d.png">

### Steps to Reproduce

This is my code.

[https://www.pixiplayground.com/#/edit/bPd-mK7kIhRrtlASgsQ93](https://www.pixiplayground.com/#/edit/bPd-mK7kIhRrtlASgsQ93)

	function draw(g){
		g.moveTo(406,8);
		g.bezierCurveTo(401,-21,289,-803,269,-804);
		g.bezierCurveTo(242,-805,229,-806,215,-806);
		g.bezierCurveTo(201,-806,187,-805,161,-804);
		g.bezierCurveTo(140,-803,18,-19,13,0);
		g.bezierCurveTo(13,1,12,3,12,4);
		g.bezierCurveTo(12,8,14,12,26,12);
		g.lineTo(149,12);
		g.bezierCurveTo(158,12,157,5,158,3);
		g.bezierCurveTo(163,-15,177,-116,182,-136);
		g.bezierCurveTo(183,-142,193,-144,203,-144);
		g.bezierCurveTo(210,-144,218,-143,223,-143);
		g.bezierCurveTo(232,-143,240,-142,241,-136);
		g.bezierCurveTo(243,-118,266,-12,271,5);
		g.bezierCurveTo(272,8,276,13,279,13);
		g.bezierCurveTo(330,13,364,14,397,14);
		g.bezierCurveTo(401,14,406,12,406,8);
		g.closePath();

		g.moveTo(232,-269);
		g.bezierCurveTo(232,-265,224,-264,214,-264);
		g.bezierCurveTo(204,-264,192,-266,193,-269);
		g.bezierCurveTo(197,-291,204,-383,213,-383);
		g.bezierCurveTo(219,-383,228,-300,232,-269);
		g.closePath();
	}

	var app = new PIXI.Application({ 
		width: 400, height: 450 ,
		antialias : true,
		backgroundColor:0xc1fcbc,
	});
	document.body.appendChild(app.view);
	const g = new PIXI.Graphics();
	g.beginFill(0xf00000 ,1);
	g.setMatrix(new PIXI.Matrix(0.5,0,0,0.5,20,400));
	this.draw(g);
	g.endFill();
	app.stage.addChild(g);

	var canvas = document.createElement('canvas');
	var ctx = canvas.getContext('2d');
	canvas.setAttribute("style","margin-left:10px;");
	canvas.width = 400;
	canvas.height = 450;
	ctx.fillStyle = "#c1fcbc";
	ctx.fillRect(0, 0, canvas.width, canvas.height);
	ctx.fillStyle = '#f00000';
	ctx.setTransform(0.5,0,0,0.5,20,400);
	ctx.beginPath();
	this.draw(ctx);
	ctx.fill();
	document.body.appendChild(canvas);

### Environment

- **`pixi.js` version**: *e.g. 7.1.0*
- **Browser & Version**: *e.g. Chrome 108*
- **OS & Version**: *e.g. Ubuntu 22.04*
- **Running Example**: *e.g. <https://www.pixiplayground.com/#/edit/bPd-mK7kIhRrtlASgsQ93/>*

### Additional Information

If you have any ideas, please let me know. Thank you! 🙏

**Comment by bigtimebuddy** [comment, marked as answer]
Try this:
https://www.pixiplayground.com/#/edit/QEkvNT7UpyXfmMORjfKle

Graphics and context2d drawing are not equivalent in how they render holes. Context2d relies on the direction of the path to determine the hole (nonzero fill rule) and Graphics is declarative with holes (beginHole, endHole) this might be called (similar to evenodd fill rule).

Hope this helps!

**Reply by ymma** [reply]
Thank you very much for your answer, it helped me to solve the problem 😄
